src/init.cpp

```diff
@@ -765,9 +765,9 @@ static std::string ResolveErrMsg(const char * const optname, const std::string&
     return strprintf(_("Cannot resolve -%s address: '%s'"), optname, strBind);
 }
 
-static std::string AmountErrMsg(const char * const optname, const std::string& strValue)
+static std::string AmountErrMsg(const std::string& optname, const std::string& strValue)
 {
-    return strprintf(_("Invalid amount for -%s=<amount>: '%s'"), optname, strValue);
+    return strprintf(_("Invalid amount for %s."), strprintf("-%s (%s=%s)", optname, optname, strValue));
```

Comment:
There is nothing wrong but I'd rather not expose three %s to translators, considering that the syntax in the braces does not change. Also, this is only a short note to the user that a specific arg can't be parsed, so not worth to have perfect grammar in every possible language, imo.

Though, I am happy to revert, if that's wanted.